src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/ILoggerWithColor.cs

```diff
@@ -15,5 +15,7 @@ internal interface ILoggerWithColor : ILogger
         /// <param name="data">The message text to be logged.</param>
         /// <param name="color">The ConsoleColor value to set the text color.</param>
         void LogMinimal(string data, ConsoleColor color);
+
+        LogLevel LogLevel { get; set; }
```

Comment:
ILoggerWithColorAndLogLevel maybe?

jk, but it might be a good idea to consider a better name for this logger. 

On a more serious note, it feels unusual to have a setter for the log level. In fact, the log level even being disposed of an anti-pattern from an OOP perspective.
 
IMO we should have a wrapper or if possible delay the logger creation.